src/monitord/monitor_actions.c

```diff
@@ -0,0 +1,256 @@
+                        /* The agent is now connected, removing from table */
+                        agent_hash_prev_node = agent_hash_node->prev;
+                        OSHash_Delete_ex(agents_to_alert_hash, agent_hash_node->key);
+                        agent_hash_node = agent_hash_prev_node;
+                    }
+
+                    else if (strcmp(j_agent_status->valuestring, "disconnected") == 0 &&
+                            j_agent_lastkeepalive->valueint < (time(0) - (mond.global.agents_disconnection_time + mond.global.agents_disconnection_alert_time))) {
+                        /* Generating the disconnection alert */
+                        char *agent_name_ip = NULL;
+                        os_strdup(j_agent_name->valuestring, agent_name_ip);
+                        wm_strcat(&agent_name_ip, j_agent_ip->valuestring, '-');
+                        monitor_send_disconnection_msg(agent_name_ip);
+                        agent_hash_prev_node = agent_hash_node->prev;
+                        OSHash_Delete_ex(agents_to_alert_hash, agent_hash_node->key);
+                        agent_hash_node = agent_hash_prev_node;
+                        os_free(agent_name_ip);
+                    }
+                }
+        } else {
+            mdebug1("Unable to retrieve agent's '%s' data from Wazuh DB", agent_hash_node->key);
+            agent_hash_prev_node = agent_hash_node->prev;
+            OSHash_Delete_ex(agents_to_alert_hash, agent_hash_node->key);
+            agent_hash_node = agent_hash_prev_node;
+        }
+        cJSON_Delete(j_agent_info);
+        agent_hash_node = OSHash_Next(agents_to_alert_hash, &inode_it, agent_hash_node);
+    }
+}
+
+void monitor_agents_deletion(){
+    int *agents_array;
+    cJSON *j_agent_info = NULL;
+    cJSON *j_agent_status = NULL;
+    cJSON *j_agent_lastkeepalive = NULL;
+    cJSON *j_agent_name = NULL;
+    cJSON *j_agent_ip = NULL;
+
+     agents_array = wdb_get_agents_by_connection_status("disconnected", &sock);
+    if (agents_array) {
+        for (int i = 0; agents_array[i] != -1; i++) {
+            j_agent_info = wdb_get_agent_info(agents_array[i], &sock);
+            if (j_agent_info) {
+                j_agent_name = cJSON_GetObjectItem(j_agent_info->child, "name");
+                j_agent_lastkeepalive = cJSON_GetObjectItem(j_agent_info->child, "last_keepalive");
+                j_agent_status = cJSON_GetObjectItem(j_agent_info->child, "connection_status");
+                j_agent_ip = cJSON_GetObjectItem(j_agent_info->child, "register_ip");
+
+                if (cJSON_IsString(j_agent_status) && j_agent_status->valuestring != NULL &&
+                    cJSON_IsString(j_agent_name) && j_agent_name->valuestring != NULL &&
+                    cJSON_IsString(j_agent_ip) && j_agent_ip->valuestring != NULL &&
+                    cJSON_IsNumber(j_agent_lastkeepalive)) {
+
+                        if (strcmp(j_agent_status->valuestring, "disconnected") == 0 &&
```

Comment:
Code style: This code block and the one in L146 has an 8 spaces tabulation instead of 4

Reply:
The indentation was corrected!